L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboardME11.cc

```diff
@@ -0,0 +1,853 @@
+  const CSCChamber* cscChamberME1a(csc_g->chamber(me1aId));
+
+  retrieveGEMPads(gemPads, gemId);
+  retrieveGEMCoPads();
+
+  const bool hasPads(!pads_.empty());
+  const bool hasCoPads(hasPads and !coPads_.empty());
+
+  // ALCT-centric matching
+  for (int bx_alct = 0; bx_alct < CSCAnodeLCTProcessor::MAX_ALCT_BINS; bx_alct++)
+  {
+    if (alct->bestALCT[bx_alct].isValid())
+    {
+      const int bx_clct_start(bx_alct - match_trig_window_size/2);
+      const int bx_clct_stop(bx_alct + match_trig_window_size/2);
+      const int bx_copad_start(bx_alct - maxDeltaBXCoPad_);
+      const int bx_copad_stop(bx_alct + maxDeltaBXCoPad_);
+
+      if (debug_matching){
+        LogTrace("CSCGEMCMotherboardME11") << "========================================================================" << std::endl
+                                           << "ALCT-CLCT matching in ME1/1 chamber: " << cscChamber->id() << std::endl
+                                           << "------------------------------------------------------------------------" << std::endl
+                                           << "+++ Best ALCT Details: " << alct->bestALCT[bx_alct]  << std::endl
+                                           << "+++ Second ALCT Details: " << alct->secondALCT[bx_alct] << std::endl;
+
+        printGEMTriggerPads(bx_clct_start, bx_clct_stop, CSCPart::ME11);
+        printGEMTriggerCoPads(bx_clct_start, bx_clct_stop, CSCPart::ME11);
+
+        LogTrace("CSCGEMCMotherboardME11") << "------------------------------------------------------------------------" << std::endl
+                                           << "Attempt ALCT-CLCT matching in ME1/b in bx range: [" << bx_clct_start << "," << bx_clct_stop << "]" << std::endl;
+      }
+
+      // ALCT-to-CLCT matching in ME1b
+      int nSuccesFulMatches = 0;
+      for (int bx_clct = bx_clct_start; bx_clct <= bx_clct_stop; bx_clct++)
+      {
+        if (bx_clct < 0 or bx_clct >= CSCCathodeLCTProcessor::MAX_CLCT_BINS) continue;
+        if (drop_used_clcts and used_clct_mask[bx_clct]) continue;
+        if (clct->bestCLCT[bx_clct].isValid())
+        {
+          const int quality(clct->bestCLCT[bx_clct].getQuality());
+          if (debug_matching) LogTrace("CSCGEMCMotherboardME11") << "++Valid ME1b CLCT: " << clct->bestCLCT[bx_clct] << std::endl;
+
+	  // pick the pad that corresponds
+	  matches<GEMPadDigi> mPads;
+	  matchingPads<GEMPadDigi>(clct->bestCLCT[bx_clct], clct->secondCLCT[bx_clct],
+				   alct->bestALCT[bx_alct], alct->secondALCT[bx_alct], ME1B, mPads);
+	  matches<GEMCoPadDigi> mCoPads;
+	  matchingPads<GEMCoPadDigi>(clct->bestCLCT[bx_clct], clct->secondCLCT[bx_clct],
+				     alct->bestALCT[bx_alct], alct->secondALCT[bx_alct], ME1B, mCoPads);
+
+          if (dropLowQualityCLCTsNoGEMs_ME1b_ and quality < 4 and hasPads){
+            int nFound(mPads.size());
+            const bool clctInEdge(clct->bestCLCT[bx_clct].getKeyStrip() < 5 or clct->bestCLCT[bx_clct].getKeyStrip() > 124);
```

Comment:
it seems like these magic numbers need to be defined someplace or better yet derived from the geometry itself. What prevents that?

Reply:
I added a comment in the code why these numbers are there. I can look for a better way to do it (but not in this PR)